dora/core/common/src/main/java/alluxio/membership/MembershipManager.java

```diff
@@ -0,0 +1,112 @@
+package alluxio.membership;
+
+import alluxio.MembershipType;
+import alluxio.conf.AlluxioConfiguration;
+import alluxio.conf.PropertyKey;
+import alluxio.resource.LockResource;
+import alluxio.wire.WorkerInfo;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantLock;
+import javax.annotation.concurrent.GuardedBy;
+
+/**
+ * Interface for worker membership management module.
+ */
+public interface MembershipManager extends AutoCloseable {
+
+  /**
+   * An idempotent call to register to join the membership.
+   * @param worker
+   * @throws IOException
+   */
+  public void join(WorkerInfo worker) throws IOException;
+
+  /**
+   * Get all registered worker members.
+   * @return all registered workers
+   * @throws IOException
+   */
+  public List<WorkerInfo> getAllMembers() throws IOException;
+
+  /**
+   * Get healthy workers.
+   * @return healthy worker list
+   * @throws IOException
+   */
+  public List<WorkerInfo> getLiveMembers() throws IOException;
```

Comment:
make sense, i will address this as part of change to address dbw9580 's comment to review a View class instead of list of workerinfos, will address this in subsequent PRs